// @yext/cobalt/src/components/AboutMap/About.tsx
import React from "react";
import { Link } from "@yext/pages/components";
import { Coordinate, CTA } from "@yext/types";
import { LocationMap } from "@yext/sites-react-components"
import { MapboxMaps } from "@yext/components-tsx-maps";
import "./About.css";

export interface AboutType {
  title?: string
  body?: string
  cta?: CTA
  coordinate: Coordinate
}

export interface AboutProps extends AboutType {
  textLeft?: boolean
}

export default function About ({
  textLeft,
  ...content
}: AboutProps) {
  const textEl = (content.title && content.body) && (
    <div className="sm:col-span-7">
      <h2 className="pb-4 sm:pb-8 text-brand-text-medium text-3xl" >{content.title}</h2>
      <p className="pb-4 sm:pb-8 text-brand-text-light">{content.body}</p>
      {content.cta && <Link className="bg-blue-500 p-2 rounded text-white" cta={content.cta} />}
    </div>
  );

  const mapEl = content.coordinate && (
    <div className="sm:col-span-5">
      <LocationMap
        provider={MapboxMaps}
        coordinate={content.coordinate}
        apiKey={"pk.eyJ1IjoieWV4dCIsImEiOiJqNzVybUhnIn0.hTOO5A1yqfpN42-_z_GuLw"}
      >
        <svg fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 40 55" width="40" height="53">
          <path d="M17.962 53.09C2.952 31.163.167 28.913.167 20.853.167 9.813 9.047.864 20 .864c10.954 0 19.833 8.95 19.833 19.988 0 8.06-2.786 10.31-17.795 32.239a2.47 2.47 0 0 1-4.076 0Z" fill="#FF6100"/>
          <path d="M5.125 22.787v5.177h3.872v-4.543c0-2.016 1.141-3.32 3.917-3.41v-3.799c-5.1.084-7.789 2.684-7.789 6.575Zm22.378-6.744c-3.594.02-6.176 2.558-6.176 6.046 0 3.445 2.153 5.706 5.788 6.046l2.54-1.882.086.042v1.67h3.894v-5.876c0-3.659-2.445-6.067-6.131-6.046Zm0 8.371c-1.334 0-2.302-.973-2.302-2.325 0-1.352.968-2.325 2.302-2.325 1.335 0 2.259.972 2.259 2.325 0 1.353-.94 2.325-2.259 2.325Zm-8.156-8.202h-3.872v11.754h3.872V16.212ZM17.41 9.541c-.51 0-1.009.149-1.433.427a2.542 2.542 0 0 0-.95 1.136 2.484 2.484 0 0 0-.146 1.463c.1.49.345.941.707 1.295.36.354.82.595 1.321.692.501.098 1.02.047 1.492-.144a2.571 2.571 0 0 0 1.158-.932 2.495 2.495 0 0 0-.321-3.197 2.585 2.585 0 0 0-.838-.548 2.625 2.625 0 0 0-.99-.192Z" fill="#fff"/>
        </svg>
      </LocationMap>
    </div>
  )

  return (
    <section className="About container mx-auto px-4 grid gap-8 items-center sm:grid-cols-12 py-8 sm:py-16">
      {textLeft ? [textEl, mapEl] : [mapEl, textEl]}
    </section>
  )
}
