// smartling-api-sdk-nodejs/built/api/jobs/params/list-jobs-parameters.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListJobsParameters = void 0;
const index_1 = require("../../parameters/index");
const index_2 = require("../../exception/index");
class ListJobsParameters extends index_1.BaseParameters {
    setName(jobName) {
        if (jobName.length >= 170) {
            throw new index_2.SmartlingException("Job name should be less than 170 characters.");
        }
        this.set("jobName", jobName);
        return this;
    }
    setLimit(limit) {
        if (limit > 0) {
            this.set("limit", limit);
        }
        return this;
    }
    setOffset(offset) {
        if (offset >= 0) {
            this.set("offset", offset);
        }
        return this;
    }
    setStatuses(statuses) {
        this.set("translationJobStatus", statuses);
        return this;
    }
    setSort(field, order) {
        this.set("sortBy", field);
        this.set("sortDirection", order);
        return this;
    }
}
exports.ListJobsParameters = ListJobsParameters;
