// @yext/cobalt/src/plopfile.mjs
import fs from 'fs';

/**
 * Plop.js is a sub-module generator. This file configures a set of CLI prompts
 * that will pull components from this repo into the repo it is called from.
 * 
 * This generator should be loaded and called from starter sites, not this repo.
 */

const getStarterTemplateList = (path) => {
  const files = fs.readdirSync(path, { withFileTypes: true });

  return files
    .filter((item) => item.isDirectory())
    .map((item) => item.name);
};

export default function (plop) {
  const ROOT_COMPONENTS = plop.getPlopfilePath() + '/components';

  plop.setGenerator('components', {
    description: 'generate starter components',

    async prompts(inquirer) {
      const queue = [{
        type: 'checkbox',
        name: 'components',
        message: 'components:',
        choices: getStarterTemplateList(ROOT_COMPONENTS),
      }];
      let answers = {};

      while (queue.length) {
        const next = await inquirer.prompt(queue.shift());

        if (next.components) {
          // Add follow-up prompts for every component being added
          next.components.forEach((component) => {
            queue.push(
              {
                type: 'input',
                name: `customName${component}`,
                message: `Name for ${component} component:`,
                default: component,
              }
            );
          });
        }

        answers = {...answers, ...next};
      }

      // Explicitly pass prompt answers to 'actions' because async prompts
      plop.answers = answers;
    },

    actions: function() {
      const actions = [];

      plop.answers.components.forEach((component) => {
        const data = fs.readFileSync(`${ROOT_COMPONENTS}/${component}/manifest.json`);
        const manifest = JSON.parse(data);
        const customName = plop.answers['customName' + component];

        manifest.files.forEach((file) => {
          const customFile = file.replace(manifest.name, customName);

          // Create files
          actions.push({
            type: 'add',
            path: `src/components/${customName}/${customFile}`,
            templateFile: `components/${component}/${file}`,
          });

          // Set custom name
          actions.push({
            type: 'modify',
            path: `src/components/${customName}/${customFile}`,
            pattern: new RegExp(manifest.name, `gm`),
            template: customName,
          });
        });

        // Add named export to components/index.ts
        actions.push({
          type: 'append',
          path: `src/components/index.ts`,
          template: manifest.export || "export { default as {{name}} } from './{{name}}/{{name}}';",
          data: {
            name: customName,
          },
        });
      });

      return actions;
    }
  });
}
