// smartling-api-sdk-nodejs/built/api/files/params/file-type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileType = void 0;
var FileType;
(function (FileType) {
    FileType["ANDROID"] = "android";
    FileType["IOS"] = "ios";
    FileType["GETTEXT"] = "gettext";
    FileType["HTML"] = "html";
    FileType["JAVA_PROPERTIES"] = "java_properties";
    FileType["XLIFF"] = "xliff";
    FileType["XML"] = "xml";
    FileType["JSON"] = "json";
    FileType["DOCX"] = "docx";
    FileType["PPTX"] = "pptx";
    FileType["XLSX"] = "xlsx";
    FileType["IDML"] = "idml";
    FileType["QT"] = "qt";
    FileType["RESX"] = "resx";
    FileType["PLAIN_TEXT"] = "plain_text";
    FileType["CSV"] = "csv";
    FileType["SRT"] = "srt";
    FileType["STRINGSDICT"] = "stringsdict";
    FileType["XLS"] = "xls";
    FileType["DOC"] = "doc";
    FileType["PPT"] = "ppt";
    FileType["PRES"] = "pres";
    FileType["MADCAP"] = "madcap";
    FileType["YAML"] = "yaml";
    FileType["MARKDOWN"] = "markdown";
})(FileType = exports.FileType || (exports.FileType = {}));
