// smartling-api-sdk-nodejs/built/api/files/params/upload-file-parameters.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UploadFileParameters = void 0;
const fs = __importStar(require("fs"));
const string_to_file_stream_1 = __importDefault(require("string-to-file-stream"));
const index_1 = require("../../parameters/index");
class UploadFileParameters extends index_1.BaseParameters {
    setDirective(key, value) {
        this.set(`smartling.${key}`, value);
        return this;
    }
    setFile(filePath) {
        this.set("file", fs.createReadStream(fs.realpathSync(filePath)));
        return this;
    }
    setFileFromLocalFilePath(filePath) {
        this.set("file", fs.createReadStream(fs.realpathSync(filePath)));
        return this;
    }
    setFileContent(fileContent) {
        this.set("file", (0, string_to_file_stream_1.default)(fileContent));
        return this;
    }
    setFileType(fileType) {
        this.set("fileType", fileType);
        return this;
    }
    setFileUri(fileUri) {
        this.set("fileUri", fileUri);
        return this;
    }
}
exports.UploadFileParameters = UploadFileParameters;
