// @yext/cobalt/src/components/AboutImage/About.tsx
import React from "react";
import { Image, Link } from "@yext/pages/components";
import { Image as ImageType, CTA } from "@yext/types";

export interface AboutType {
  title?: string
  body?: string
  cta?: CTA
  image?: ImageType
}

export interface AboutProps extends AboutType {
  textLeft?: boolean
}

export default function About ({
  textLeft,
  ...content
}: AboutProps) {
  const textEl = (content.title && content.body) && (
    <div className="sm:col-span-7">
      <h2 className="pb-4 sm:pb-8 text-brand-text-medium text-3xl" >{content.title}</h2>
      <p className="pb-4 sm:pb-8 text-brand-text-light">{content.body}</p>
      {content.cta && <Link className="bg-blue-500 p-2 rounded text-white" cta={content.cta} />}
    </div>
  );

  const imageEl = content.image && (
    <Image
      image={content.image}
      className="w-full sm:col-span-5"
    />
  )

  return (
    <section className="About container mx-auto px-4 grid gap-8 items-center sm:grid-cols-12 py-8 sm:py-16">
      {textLeft ? [textEl, imageEl] : [imageEl, textEl]}
    </section>
  )
}
