// @yext/cobalt/src/components/AboutImage/About.stories.tsx
import React from "react";
import { ComponentStory, ComponentMeta } from '@storybook/react';

import About, { AboutProps } from "./About";

// Sample data

const IMAGE = {
  height: 283,
  url: 'https://a.mktgcdn.com/p/E5tDjnkkrNPQWJHG3Z2vCEZ7GlEjo3lU6IZVnzIWkP0/405x283.png',
  width: 405
};

const LEARN_MORE_CTA = {
  link: 'https://yext.com/',
  label: 'Learn More',
}

export default {
  title: 'Sections/About',
  component: About,
  parameters: {
    layout: 'fullscreen',
  },
} as ComponentMeta<typeof About>;

const Template: ComponentStory<typeof About> = (args) => (
  <About {...args} />
);

// More on interaction testing: https://storybook.js.org/docs/react/writing-tests/interaction-testing
export const AboutImage: ComponentStory<React.FC<AboutProps>> = Template.bind({});
AboutImage.args = {
  title: "Yext: One Platform. Limitless solutions.",
  body: 'Deliver relevant, actionable answers wherever your customers, employees, and partners look for information.',
  cta: LEARN_MORE_CTA,
  image: IMAGE,
};
