// @yext/cobalt/src/components/Promo/Promo.tsx
import React from "react";
import c from "classnames";
import { Image as ImageType, CTA } from "@yext/types";
import { Image, Link } from "@yext/pages/components";

/**
 * Subset of data coming from platform
 */
export interface PromoType {
  title: string
  body: string
  image?: ImageType
  cta?: CTA
}

export interface PromoProps extends PromoType {
  textLeft?: boolean
}

export default function Promo ({
  textLeft,
  ...content
}: PromoProps) {
  const textEl = (content.title && content.body) && (
    <div className="sm:col-span-7">
      <h2 className="text-3xl pb-4 sm:pb-8">
        {content.title}
      </h2>
      <p className="pb-4 sm:pb-8">{content.body}</p>
      {content.cta && (
        <Link
          className="bg-blue-500 p-2 rounded text-white"
          cta={content.cta}
        />
      )}
    </div>
  );

  const imageEl = content.image && (
    <Image
      image={content.image}
      className="w-full sm:col-span-5"
    />
  );

  return (
    <section className="container mx-auto px-4 grid gap-8 items-center sm:grid-cols-12 py-8 sm:py-16">
      {textLeft ? [textEl, imageEl] : [imageEl, textEl]}
    </section>
  );
}
