// @yext/cobalt/src/components/Promo/Promo.stories.tsx
import React from "react";
import { ComponentStory, ComponentMeta } from '@storybook/react';

import Promo, { PromoProps } from "./Promo";

// Sample data

const IMAGE = {
  height: 283,
  url: 'https://a.mktgcdn.com/p/E5tDjnkkrNPQWJHG3Z2vCEZ7GlEjo3lU6IZVnzIWkP0/405x283.png',
  width: 405
};

const LEARN_MORE_CTA = {
  link: 'https://yext.com/',
  label: 'Learn More',
}

export default {
  title: 'Sections/Promo',
  component: Promo,
  parameters: {
    layout: 'fullscreen',
  },
} as ComponentMeta<typeof Promo>;

const Template: ComponentStory<typeof Promo> = (args) => (
  <Promo {...args} />
);

// More on interaction testing: https://storybook.js.org/docs/react/writing-tests/interaction-testing
export const SinglePromo: ComponentStory<React.FC<PromoProps>> = Template.bind({});
SinglePromo.args = {
  title: "New customer $0 fee promotion",
  body: 'New customer offer only. Use code HelloRia on the Ria Money Transfer app or RiaMoneyTransfer.com and receive $0 fees on first time money transfers of $50 or more, limited to one (1) per person. Offer expires on December 31, 2022. Ria Money Transfer reserves the right to terminate the promotion at any time.',
  image: IMAGE,
  cta: LEARN_MORE_CTA,
};
